### PR TITLE
Related to  #138, add package scanner and .travis.yml

### PR DESCRIPTION
**Pull request by FedericoCeratto**

### CODE REVIEW

**Comment by flaviut** [comment]
LGTM

/cc @dom96 Lets get this merged?

**Comment by dom96** [comment]
I would need to set up travis for this I guess?

**Comment by FedericoCeratto** [comment]
@dom96: enable TravisCI on GitHub? Yes.
Setting up a local Travis instance on your dev host? No

**Comment by fowlmouth** [comment]
@dom96 is there a travis-ci account for nim-lang if not I can set up one up.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+os:
+  - linux
+
+language: c
+
+env:
+  - BRANCH="travis"
+  - BRANCH="devel"
+
+install:
```
Comment (megawac) [inline .travis.yml]: Probably should be `before_install`; `install` would generally cover things like `nimble install` or `nimble build`

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+
+os:
+  - linux
+
+language: c
+
+env:
+  - BRANCH="travis"
+  - BRANCH="devel"
```
Comment (megawac) [inline .travis.yml]: Why do you want to setup a matrix?

File: package_scanner.nim [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    elif not (pdata["method"].str in VCS_TYPES):
+      echo "E: ", name, "has an unknown method: ", pdata["method"].str
+      result.inc()
+
+    if not pdata.hasKey("license"):
+      echo "E: ", name, "has no license"
+      result.inc()
+    elif not (pdata["license"].str in LICENSES):
+      echo "W: ", name, "has an unexpected license: ", pdata["license"]
```
Comment (megawac) [inline package_scanner.nim]: Is there a good reason to be so strict with licencing as to fail a build if it doesn't appear in that list? For example you're missing `MPL` which I'd say is entirely valid. Also is case sensitivity important in a license?

File: package_scanner.nim [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+      except HttpRequestError, TimeoutError, AssertionError:
+        echo "E: ", name, ": unable to fetch repository ", url, " ", getCurrentExceptionMsg()
+        result.inc()
+
+  echo "Error count: ", result
+  return
+
+when isMainModule:
+  quit(check())
```
Comment (megawac) [inline package_scanner.nim]: Will this generate the right exit code? I think travis will only fail for `0` responses

Comment (flaviut) [inline package_scanner.nim]: The exact opposite actually:
>All commands must exit with code 0 on success. Anything else is considered failure.

Comment (megawac) [inline package_scanner.nim]: haha I just realized that

File: package_scanner.nim (hunk 3 above)
Comment (flaviut) [inline package_scanner.nim]: The array might be incomplete, but having canonical names for licenses is important for tooling.

**Pull request merged**